raiden/raiden_event_handler.py

```diff
@@ -114,176 +124,198 @@ def unlock(
-            self.handle_send_withdrawrequest(raiden, event)
-        elif type(event) == SendWithdrawConfirmation:
-            assert isinstance(event, SendWithdrawConfirmation), MYPY_ANNOTATION
-            self.handle_send_withdraw(raiden, event)
-        elif type(event) == SendWithdrawExpired:
-            assert isinstance(event, SendWithdrawExpired), MYPY_ANNOTATION
-            self.handle_send_withdrawexpired(raiden, event)
-        elif type(event) == SendProcessed:
-            assert isinstance(event, SendProcessed), MYPY_ANNOTATION
-            self.handle_send_processed(raiden, event)
-        elif type(event) == EventPaymentSentSuccess:
-            assert isinstance(event, EventPaymentSentSuccess), MYPY_ANNOTATION
-            self.handle_paymentsentsuccess(raiden, event)
-        elif type(event) == EventPaymentSentFailed:
-            assert isinstance(event, EventPaymentSentFailed), MYPY_ANNOTATION
-            self.handle_paymentsentfailed(raiden, event)
-        elif type(event) == EventUnlockFailed:
-            assert isinstance(event, EventUnlockFailed), MYPY_ANNOTATION
-            self.handle_unlockfailed(raiden, event)
-        elif type(event) == EventInvalidSecretRequest:
-            assert isinstance(event, EventInvalidSecretRequest), MYPY_ANNOTATION
-            self.handle_invalidsecretrequest(raiden, event)
-        elif type(event) == ContractSendSecretReveal:
-            assert isinstance(event, ContractSendSecretReveal), MYPY_ANNOTATION
-            self.handle_contract_send_secretreveal(raiden, event)
-        elif type(event) == ContractSendChannelClose:
-            assert isinstance(event, ContractSendChannelClose), MYPY_ANNOTATION
-            self.handle_contract_send_channelclose(raiden, chain_state, event)
-        elif type(event) == ContractSendChannelUpdateTransfer:
-            assert isinstance(event, ContractSendChannelUpdateTransfer), MYPY_ANNOTATION
-            self.handle_contract_send_channelupdate(raiden, event)
-        elif type(event) == ContractSendChannelBatchUnlock:
-            assert isinstance(event, ContractSendChannelBatchUnlock), MYPY_ANNOTATION
-            self.handle_contract_send_channelunlock(raiden, chain_state, event)
-        elif type(event) == ContractSendChannelSettle:
-            assert isinstance(event, ContractSendChannelSettle), MYPY_ANNOTATION
-            self.handle_contract_send_channelsettle(raiden, event)
-        elif type(event) == ContractSendChannelWithdraw:
-            assert isinstance(event, ContractSendChannelWithdraw), MYPY_ANNOTATION
-            self.handle_contract_send_channelwithdraw(raiden, event)
-        elif type(event) in UNEVENTFUL_EVENTS:
-            pass
-        else:
-            log.error(
-                "Unknown event",
-                event_type=str(type(event)),
-                node=to_checksum_address(raiden.address),
-            )
+        message_queues: Dict[QueueIdentifier, List[Message]] = defaultdict(list)
```

Comment:
This is a bit of a hack, but it is necessary ATM because the event handler is not separated between smart contract function calls and messages.

In the future we should have one event handler for each type of event, and then the code can be simplified.

Reply:
that is sort of the idea, I didn't want to change in this PR because it was large enough and it would just be a bit better, but not improve performance